### PR TITLE
Remove FakeFunc from AddPdf/AddPdfNorm._parts

### PR DESCRIPTION
**Pull request by marinang**
This is a workaround that worked for me and Following #84, I think it will solve a couple of tests.

### CODE REVIEW

**Comment by marinang** [comment]
So since there were few remaining test failures I tried to fix them as well.

**Comment by marinang** [comment]
So the builds work for python 2.7 and 3.4, and it works for me for python 3.7. 

The only two tests that are remaining are two plot check that fails and I cannot see the difference to solve.

**Comment by eduardo-rodrigues** [comment]
Hi @marinang, thumbs up for the heroic effort!
I have not yet looked at the `test_plottting.py` file, which is where the failing tests come from, but one thing is really strange: at first all 81 test pass, ran with command `python -m pytest -v`. Then Travis seems to be rerunning all of `test_plotting.py` with the command `python -m pytest -v --mpl tests/test_plotting.py`, and there 2 tests fail. Anything obvious?

**Comment by marinang** [comment]
if you look at the makefile.

**Comment by eduardo-rodrigues** [comment]
Yep, got there as well, actually looking at the various files needed. While doing that I also noticed the comments

> There is a slight difference in the x-axis tick label positioning for this
> plot between Python 2 and 3, it's not important here so increase the RMS
> slightly such that it's ignored

in a couple of tests in tests_plotting.py where a tolerance is specified. Could it be a matter of increasing the tolerance for the 2 failing tests?

**Comment by marinang** [comment]
@eduardo-rodrigues you have to put a big tolerance because the rms are ~45 and 56 for between the expected and obtained images.
I put a big tolerance of 60 for those two test for py35 and py36, to make the test pass.

Two remaining problems still for py35 and py36, the doc is not compiled because of this module 'matplotlib.sphinxext.only_directives' which seems to not exist anymore.

**Comment by eduardo-rodrigues** [comment]
Saw that. It's a weird thing the differences between Py 3.4 and 3.5-3.6. To be revisited at some point it seems …

I am tempted to accept this PR given that what is addressed is sorted-ish. The failure at creating the documentation is a different matter and deserves its own PR I would say. I browsed the web a fit and may try and address the problem with the doc soon.

I will merge this later unless somebody shouts.

**Comment by marinang** [comment]
@eduardo-rodrigues  I just tried to set a maximum requirement on the matplotlib version in travis to the public one, because from the log I saw that the mil version was 3.0.1 ... so just a check

File: tests/test_plotting.py [hunk 1]
```diff
@@ -16,13 +16,19 @@
...
 from probfit.plotting import draw_pdf, draw_compare_hist
 from probfit.pdf import gaussian, linear
 from probfit.funcutil import rename
 from probfit.functor import Extended, AddPdfNorm, AddPdf
 from probfit.costfunc import UnbinnedLH, BinnedLH, BinnedChi2, Chi2Regression, \
     SimultaneousFit
 
+major, minor = sys.version_info[0:2]
+if major >= 3 and minor >= 5:
+    special_tol = 60.0
```
[the remark this answers is not in the document]
Comment (marinang) [reply tests/test_plotting.py]: Have a look here https://travis-ci.org/scikit-hep/probfit/jobs/448471653.

**Comment by marinang** [comment]
@eduardo-rodrigues for the doc according to https://matplotlib.org/3.0.0/devel/plot_directive.html?highlight=sphinx#module-matplotlib.sphinxext.plot_directive it looks like matplotlib.sphinxext.only_directive doesn't exist anymore.

**Comment by eduardo-rodrigues** [comment]
Yes, seems the directive is gone with matplotlib >=3.

**Pull request merged**